### PR TITLE
Added system test cases - Initial tests

### PR DESCRIPTION
**Pull request by sunilkumarn417**
Signed-off-by: sunilkumarn417 <sunnagar@redhat.com>

Added mixed system test cases.

### CODE REVIEW

File: qa/tasks/mixed_system_test.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    """
+    daemons = config.get('daemons')
+    try:
+        for daemon in daemons:
+            stats = system.get_daemon_info(daemon, ctx)
+            assert stats.get("active_count") > 0,\
+                "{} Not found in cluster".format(daemon)
+            assert system.ceph_daemon_system_test(ctx, daemon)
+            log.info("{} completed".format(daemon))
+            sleep(60)
```
Comment (mykaul) [inline qa/tasks/mixed_system_test.py]: What's the point of this sleep?

File: qa/tasks/mixed_system_tests/ios.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+    def __enter__(self):
+        log.info('starting rgw-tests')
+        log.info('config %s' % self.config)
+        if self.config is None:
+            self.config = {}
+        assert isinstance(self.config, dict), \
+            "task set-repo only supports a dictionary for configuration"
+        config_file_name = self.config['test'] + ".yaml"
+        log.info('test_version: %s' % self.config.get('test_version', 'v2'))
```
Comment (mykaul) [inline qa/tasks/mixed_system_tests/ios.py]: All those general info logs on the test version and such should most likely go to the parent class.

File: qa/tasks/mixed_system_tests/ios.py [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+            map(self.cleanup, self.soot)
+            remote.run(args=['mkdir', test_root_dir])
+            log.info('cloning the repo to %s' % remote.hostname)
+            remote.run(
+                args=[
+                    'cd',
+                    '%s' % test_root_dir,
+                    run.Raw(';'),
+                    'git',
+                    'clone',
```
Comment (mykaul) [inline qa/tasks/mixed_system_tests/ios.py]: You can probably do a shallow clone, which will save some time ('--depth 1' )

File: qa/tasks/mixed_system_tests/ios.py [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+        config_file = os.path.join(cephqe_scripts['REPO_NAME'],
+                                   cephqe_scripts['DIR'][self.config.get('test_version', 'v2')]['config'],
+                                   config_file_name)
+        log.info('script: %s' % script)
+        log.info('config_file: %s' % config_file)
+        self.soot = [test_venv, test_root_dir, 'io_info.yaml', '*.json', 'Download.*',
+                     'Download', '*.mpFile', 'x*', 'key.*', 'Mp.*', '*.key.*']
+        self.cleanup = lambda x: remote.run(args=[run.Raw('sudo rm -rf %s' % x)])
+        log.info('listing all clients: %s' % self.config.get('clients'))
+        for role in self.config.get('clients', ['client.0']):
```
Comment (mykaul) [inline qa/tasks/mixed_system_tests/ios.py]: Are you cloning into every client, serially? Why not clone once and scp to all of them? Not to mention that half of this code could have been written more easily in Ansible?

File: qa/tasks/mixed_system_test.py (hunk 1 above)
Comment (sunilkumarn417) [reply qa/tasks/mixed_system_test.py]: code has been updated and  taken out sleep.

**Comment by vasukulkarni** [comment]
Overall nice work!, Lets run in octo lab and see the logs, there are few comments above that you can address.

**Comment by sunilkumarn417** [comment]
updating new changes, will raise new PR

**Pull request closed without merging**